semgrep/semgrep/core_runner.py

```diff
@@ -426,9 +426,14 @@ def handle_regex_patterns(
             ]
         except re.error as err:
             raise SemgrepError(f"invalid regular expression specified: {err}")
-        re_fn = functools.partial(get_re_matches, patterns_re)
-        with multiprocessing.Pool(self._jobs) as pool:
-            matches = pool.map(re_fn, targets)
+
+        if len(targets) > 1:
```

Comment:
This is required because we cannot do a `multiprocessing.Pool` inside another `Pool`. We need to avoid the double `Pool` with `--test` using `Pool` and our rules in `semgrep-rules` using `pattern-regex` (uses `Pool` here). Since our tests only have one corresponding target file we can limit our `multiprocessing` use to > 1 target.

Reply:
https://stackoverflow.com/questions/6974695/python-process-pool-non-daemonic

Reply:
Have you looked into using a thread pool instead of a process pool? We got around most of these nasty issues in Bento by using multithreading inside a single process.

This won't work, of course, if your tests are mutating global state.

Reply:
cf. http://lucasb.eyer.be/snips/python-thread-pool.html

Reply:
Hmm, interesting. I'm going to save that for a future improvement. I've mostly achieved what I set out to with this, so I'm going to push it over the line :+1: